### ISSUE TITLE
Issues with duplicate files

### ISSUE THREAD

**Issue by korrat** [michaeladler/notmuch-mailmover#98]
First of all, thank you for this great project!

I've noticed that when I have duplicate files for a message, only one of the files is moved to the destination directory and the remaining files stay where they are. Could this be adjusted to move all files into the target directory?

**Comment by michaeladler** [comment]
Hi @korrat,

thank you for reporting the issue. Could you clarify what you mean by duplicate files? Could you possibly provide an example?

**Comment by korrat** [comment]
I recently sent an email and CCed myself. So, I had two files for the same message. However, `notmuch-mailmover` only moved one of these files into the target directory. The other file remained in the Sent folder, even on following invocations.

**Comment by michaeladler** [comment]
Ah, okay, I think I understand now. I don't believe there's much I can do because, technically, these are two different files (i.e., they have different filenames). Could you share the rule that you used? Perhaps the rule can be tweaked so that both files are detected. You can debug your query using the CLI with a command like:

```
notmuch search --output=files --from:foo@bar.com
```

**Comment by korrat** [comment]
If I run the command (for the specific thread), I get all three files (original message + 2 copies of the reply).
```console
> notmuch search --output=files 'thread:000000000000218a'
/home/korrat/Maildir/tsv/Sent/cur/1716530705.333818110009483e416a9835d7b68aa8.atlas,U=11:2,S
/home/korrat/Maildir/tsv/Archive/cur/b131e75f-0944-40a7-a7eb-de82ec128627,U=26:2,S
/home/korrat/Maildir/tsv/Archive/cur/34583611-969b-417f-91d9-6bdabf2158bc,U=24:2,RS
```

If I use the `notmuch-mailmover` query, I get no results:
```console
> notmuch search --output=files 'NOT folder:<target> AND thread:000000000000218a'
```

According to the notmuch documentation, both `folder` nor `path` match if _any_ copy of a message is in the directory. Therefore, notmuch-mailmover does not detect the misfiled email.

**Comment by michaeladler** [comment]
I think your issue is with `notmuch` queries and not with `notmuch-mailmover` because under the hood, `notmuch-mailmover` has a very simple logic and just forwards your query to notmuch to get back the filenames: https://github.com/michaeladler/notmuch-mailmover/blob/f38e264c876233c6ce62a98e8fc35d943e51a81d/src/engine.rs#L54

**Comment by korrat** [comment]
I think the issue is a few lines above:

https://github.com/michaeladler/notmuch-mailmover/blob/f38e264c876233c6ce62a98e8fc35d943e51a81d/src/engine.rs#L49

With that, I think my scenario looked like this:

1. I sent an email from korrat@a.com to korrat@b.com.
2. My email client copied the message to `a/sent`.
3. I manually tagged the email as `a` and `sent`.
4. I ran notmuch-mailmover (in a pre-new hook), which moved the message to `a/archive` as per the query.
5. I ran mbsync (still as part of pre-new), which downloaded a second copy of the message (same message id) to `b/inbox`.
6. On any further invocation to `notmuch-mailmover`, that second copy stays in `b/inbox`, because another copy is already in the correct folder, so the query doesn't match.

So, based on this, it seems that it comes down to some kind of race condition between steps 3 and 5 in the scenario. If I ran step 5 before tagging the email, both files would have been moved to `a/archive`.
